diesel/src/query_builder/update_statement/mod.rs

```diff
@@ -51,6 +51,11 @@ impl<T, U, V, DB> QueryFragment<DB> for UpdateStatement<T, U, V> where
     V: changeset::Changeset<DB>,
 {
     fn to_sql(&self, out: &mut DB::QueryBuilder) -> BuildQueryResult {
+        if self.values.is_noop() {
+            return Err("There are no changes to save. This \
+                       query cannot be built".into())
```

Comment:
Interesting idea. We'd need to run it through a query builder, but we have the debug query builder for that. I'm going to do it as a future refactoring.